PredicateBoolean doesn't implement Predicate<T> for T: ?Sized

<!-- Insert short summary -->

I tried this code:
```rust
let p: Box<dyn Predicate<str>> = Box::new(always());
```
I expected to see this happen: It should compile fine

Instead this happened: Build fails with this error:
```
error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> src/constant.rs:86:38
   |
86 |     let p: Box<dyn Predicate<str>> = Box::new(always());
   |                                      ^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because of the requirements on the impl of `predicates_core::core::Predicate<str>` for `constant::BooleanPredicate`
   = note: required for the cast to the object type `dyn predicates_core::core::Predicate<str>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
```

## Meta

predicates-rs version: 1.0.1
`rustc --version --verbose`:
```
> rustc --version --verbose
rustc 1.41.0-nightly (1bd30ce2a 2019-11-15)
binary: rustc
commit-hash: 1bd30ce2aac40c7698aa4a1b9520aa649ff2d1c5
commit-date: 2019-11-15
host: x86_64-unknown-freebsd
release: 1.41.0-nightly
LLVM version: 9.0
```

Originally reported at https://github.com/asomers/mockall/issues/78

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased] - Unreleased
+### Fixed
+- `BooleanPredicate` now implements `Predicate<T>` where `T: ?Sized`
+  ([#84](https://github.com/assert-rs/predicates-rs/pull/84))
 
 ## [1.0.1] - 2019-04-22
 ### Changed

--- a/src/constant.rs
+++ b/src/constant.rs
@@ -22,7 +22,7 @@ pub struct BooleanPredicate {
     retval: bool,
 }
 
-impl<Item> Predicate<Item> for BooleanPredicate {
+impl<Item: ?Sized> Predicate<Item> for BooleanPredicate {
     fn eval(&self, _variable: &Item) -> bool {
         self.retval
     }

